index.js

```diff
@@ -110,10 +112,12 @@ class Collection {
     return this.collection;
   }
 
-  static getCollectionBySlug(client, slug, params) {
+  static getCollectionBySlug(client, slug, params, options = {}) {
+    const {loadNestedCollections = true, depth = DEFAULT_DEPTH} = options;
     return client
       .getCollectionBySlug(slug, params)
-      .then(response => response && Collection.build(response["collection"] || response));
+      .then(response => response && Collection.build(response["collection"] || response))
+      .then(collection => loadNestedCollections ? loadNestedCollectionData(client, collection, { depth }) : collection);
```

Comment:
You've changed the return type of this function from an object of type `Collection` to a simple javascript object